addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/NetworkHandlerFactory.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.network.internal;
+
+import static org.openhab.binding.network.NetworkBindingConstants.*;
+
+import java.util.Map;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.network.handler.NetworkHandler;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * The handler factory retrieves the binding configuration and is responsible for creating
+ * PING_DEVICE and SERVICE_DEVICE handlers.
+ *
+ * @author David Graeff
+ */
+@Component(immediate = true, service = ThingHandlerFactory.class, name = "NetworkHandlerFactory")
+public class NetworkHandlerFactory extends BaseThingHandlerFactory {
+    final NetworkBindingConfiguration configuration = new NetworkBindingConfiguration();
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    // The activate component call is used to access the bindings configuration
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> config) {
+        super.activate(componentContext);
+        modified(config);
+    };
+
+    @Deactivate
+    protected void deactivate() {
+        super.deactivate(null);
+    }
+
+    @SuppressWarnings("null")
```

Comment:
I get those warnings a lot since the introduction if the nullable annotations, but because I use the warnings/errors tab in eclipse to find actual problems, those warnings are distracting. I therefore added some SuppressWarnings here and there.